python-manual/modules/ROOT/pages/migration.adoc

```diff
@@ -0,0 +1,333 @@
+[%collapsible]
+====
+The driver is compatible with any Python version starting from 3.6 up to 3.12.
+====
+|
+.`asyncio` support
+[%collapsible]
+====
+For using the driver's features in parallel via the link:https://docs.python.org/3/library/asyncio.html[asyncio] package.
+
+See xref:concurrency.adoc[].
+====
+
+|
+.`Driver.execute_query()` to run transactions with less knowledge of driver's internals
+[%collapsible]
+====
+The new function is a wrapper for `Session.execute_read/write()`, but it abstracts away the result processing part and returns a list of records to the caller directly.
+
+See xref:query-simple.adoc[].
+====
+|
+.Rust extension for performance
+[%collapsible]
+====
+The link:https://github.com/neo4j-drivers/neo4j-python-driver-rust-ext[Rust extension to the Python driver] is an alternative driver package that yields a 3x to 10x speedup compared to the regular driver.
+You can install it with `pip install neo4j-rust-ext`, either alongside the `neo4j` package or as a replacement to it.
+Usage-wise, the drivers are identical.
+====
+
+|
+.Export to Pandas Dataframe
+[%collapsible]
+====
+`Result.to_df()` allows for exporting a `Result` object into a Pandas DataFrame.
+
+See xref:transformers.adoc#_transform_to_pandas_dataframe[Manipulate query results -> Transform to Pandas Dataframe].
+====
+|
+.Type hints for all public APIs
+[%collapsible]
+====
+All public objects have type annotations via the link:https://docs.python.org/3/library/typing.html[typing] package. For (optional) use with type checkers and linters.
+====
+
+|
+.Re-authentication
+[%collapsible]
+====
+Allows for handling expiring authentication (backwards compatible) as well as session scoped authentication.
+
+See link:https://github.com/neo4j/neo4j-python-driver/discussions/921[Re-authentication].
+====
+|
+.Mutual TLS (mTLS) as second authentication factor (2FA)
+[%collapsible]
+====
+Allows for configuring client side TLS certificates to authenticate against the server.
+
+See link:https://github.com/neo4j/neo4j-python-driver/discussions/1031[Mutual TLS].
+====
+
+|
+.`BookmarkManager` support
+[%collapsible]
+====
+Bookmark managers make it easier to achieve causal chaining of sessions.
+
+See xref:bookmarks.adoc[].
+====
+|
+.Notification filtering API
+[%collapsible]
+====
+Filtering allows to receive only a subset of notifications from the server, and to improve performance server-side.
+
+See xref:result-summary.adoc#_filter_notifications[Explore the query execution summary -> Filter notifications].
+====
+
+|
+.GQL statuses of queries in the ResultSummary
+[%collapsible]
+====
+Eventually, this API is planned to supersede the current notifications API.
+
+See link:https://github.com/neo4j/neo4j-python-driver/discussions/1067[GQL statuses in ResultSummary].
+====
+|
+.More convenient handling of server-side warnings and notifications
+[%collapsible]
+====
+You can configure the driver to emit warnings on notifications and/or to log them.
```

Comment:
Not only that, by default, server-side warnings will be emitted as Python warning via `warning.warn(...)`. That might make a difference for people running in environments with warnings as errors.

Reply:
I copy-pasted this from the changelog without digging at all. I'll check.